### PR TITLE
Documentation for configurable merge policy

### PR DESCRIPTION
**Pull request by rishabhmaurya**
### Description
Configurable merge policy documentation
### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/5119

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by rishabhmaurya** [comment]
@hdhalter @reta please review, thank you!

**Review by reta — APPROVED** [review]

**Comment by rishabhmaurya** [comment]
@hdhalter @Naarcha-AWS can we get it merged if it looks good to you?

File: _api-reference/cluster-api/cluster-settings.md [hunk 1]
```diff
@@ -114,6 +114,7 @@ The following request field parameters are compatible with the cluster API.
...
 | remote_store.moving_average_window_size | Integer | The moving average window size used to calculate the rolling statistic values exposed through the [Remote Store Stats API]({{site.url}}{{site.baseurl}}/tuning-your-cluster/availability-and-recovery/remote-store/remote-store-stats-api/). Default is `20`. Minimum enforced is `5`. |
+| indices.time_series_index.default_index_merge_policy | String | This setting allows to specify the default merge policy for time-series indexes, particularly those with a `@timestamp` field such as data streams. The two available options are: `tiered` (default) and `log_byte_size`. `log_byte_size` is recommended for time-series indexes to enhance performance on range queries on `@timestamp` field. To override the merge policy on a per-index basis, the `index.merge.policy` index setting can be used. |
```
Comment (hdhalter) [inline _api-reference/cluster-api/cluster-settings.md:117]:
```suggestion
| indices.time_series_index.default_index_merge_policy | String | This setting allows you to specify the default merge policy for time-series indexes, particularly for those with an `@timestamp` field such as data streams. The two available options are: `tiered` (default) and `log_byte_size`.  We recommend `log_byte_size` for time-series indexes to enhance the performance of range queries with the `@timestamp` field. To override the merge policy on a per-index basis, you can use the `index.merge.policy` index setting. |
```

File: _im-plugin/index-settings.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ index.routing_partition_size | The number of shards a custom routing value can g
 index.soft_deletes.retention_lease.period | The maximum amount of time to retain a shard's history of operations. Default is `12h`.
 index.load_fixed_bitset_filters_eagerly | Whether OpenSearch should preload cached filters. Available options are `true` and `false`. Default is `true`.
 index.hidden | Whether the index should be hidden. Hidden indexes are not returned as part of queries that have wildcards. Available options are `true` and `false`. Default is `false`.
+index.merge.policy | This setting controls the merge policy for the Lucene segments. Available options are `tiered` and `log_byte_size`. Default is `tiered`. However, for time-series data, such as log events, it is advisable to switch to the `log_byte_size` merge policy. This can improve the query performance when conducting range queries on the `@timestamp` field.  It is advisable not to change the merge policy of an existing index. Instead, configure this setting when creating a new index.
```
Comment (hdhalter) [inline _im-plugin/index-settings.md:45]:
```suggestion
index.merge.policy | This setting controls the merge policy for the Lucene segments. Available options are `tiered` and `log_byte_size`. The default is `tiered`, however, for time-series data such as log events, we recommend that you use the `log_byte_size` merge policy which can improve the query performance when conducting range queries on the `@timestamp` field.  We recommend that you not change the merge policy of an existing index. Instead, configure this setting when creating a new index.
```

**Review by hdhalter — COMMENTED** [review]
Please review my changes (mostly to use active voice) to make sure I haven't changed the meaning. Thanks!

**Comment by rishabhmaurya** [comment]
Looks good, thank you @hdhalter

**Review by hdhalter — APPROVED** [review]

**Comment by hdhalter** [comment]
I submitted it for editorial review, so we can merge it after it's been reviewed by Nathan.

File: _im-plugin/index-settings.md [hunk 3]
```diff
@@ -42,6 +42,7 @@ index.routing_partition_size | The number of shards a custom routing value can g
 index.soft_deletes.retention_lease.period | The maximum amount of time to retain a shard's history of operations. Default is `12h`.
 index.load_fixed_bitset_filters_eagerly | Whether OpenSearch should preload cached filters. Available options are `true` and `false`. Default is `true`.
 index.hidden | Whether the index should be hidden. Hidden indexes are not returned as part of queries that have wildcards. Available options are `true` and `false`. Default is `false`.
+index.merge.policy | This setting controls the merge policy for the Lucene segments. Available options are `tiered` and `log_byte_size`. The default is `tiered`, however, for time-series data such as log events, we recommend that you use the `log_byte_size` merge policy which can improve the query performance when conducting range queries on the `@timestamp` field.  We recommend that you not change the merge policy of an existing index. Instead, configure this setting when creating a new index.
```
Comment (natebower) [inline _im-plugin/index-settings.md:45]:
```suggestion
index.merge.policy | This setting controls the merge policy for the Lucene segments. The available options are `tiered` and `log_byte_size`. The default is `tiered`, but for time-series data, such as log events, we recommend that you use the `log_byte_size` merge policy, which can improve query performance when conducting range queries on the `@timestamp` field. We recommend that you not change the merge policy of an existing index. Instead, configure this setting when creating a new index.
```

**Review by natebower — COMMENTED** [review]
@rishabhmaurya Please see my changes and let me know if you have any questions. Thanks!

**Comment by rishabhmaurya** [comment]
Looks good @natebower. Thank you.

**Pull request merged**